### PR TITLE
use kubebuilder go/v4 template

### PR DESCRIPTION
**Pull request by llamerada-jp**
Update kubebuilder template to go/v4 from go/v3. However, no code is moved to the cmd, internal, and test directories. This would be done in a different PR to avoid conflicts with other PRs. And the resolutions to lint's point will be addressed after the directory move is complete.

a part of: https://github.com/topolvm/topolvm/issues/760

### CODE REVIEW

**Review by peng225 — COMMENTED** [review]
In [this commit message](https://github.com/topolvm/topolvm/pull/830/commits/3d0c79741596a9659a379754fe40739f3b1da1cd), you say as follows:

> And since this project does not use custom
resources, generate-api is not generated either. Therefore, the
related codes are removed.

However, TopoLVM uses LogicalVolume custom resource. Is that OK to remove `generate-api` target?

**Review by cupnes — COMMENTED** [review]
Other than what @peng225 pointed out, there seems to be no problem.

**Comment by llamerada-jp** [comment]
@peng225 @cupnes 
The change was wrong and generate-api has been reverted.

File: go.mod [hunk 1]
```diff
@@ -8,29 +8,31 @@ require (
...
 	github.com/spf13/viper v1.10.1
 	golang.org/x/sys v0.15.0
 	google.golang.org/grpc v1.58.3
 	google.golang.org/grpc/cmd/protoc-gen-go-grpc v1.2.0
 	google.golang.org/protobuf v1.31.0
-	k8s.io/api v0.27.3
-	k8s.io/apimachinery v0.27.3
-	k8s.io/client-go v0.27.3
-	k8s.io/klog/v2 v2.90.1
+	k8s.io/api v0.28.0
```
Comment (peng225) [inline go.mod:23]: api version is upgraded to v0.28.0. IIUC, This means that the supported Kubernetes version will be changed. 

I thought this should be done in [the Kubernetes upgrade task](https://github.com/topolvm/topolvm/issues/736).

Comment (llamerada-jp) [reply go.mod:23]: I work on the k8s 1.28 support before doing the merge of this PR. After that, I will rebase.
https://github.com/topolvm/topolvm/issues/736

Comment (llamerada-jp) [reply go.mod]: I have rebased this PR. Would you review it again?

**Review by cupnes — APPROVED** [review]

**Review by peng225 — APPROVED** [review]

**Pull request merged**